### ISSUE TITLE
Missing Item/Block IDs followed by possible level.dat corruption

### ISSUE THREAD

**Issue by zcorcoran** [BeetoGuy/BetterWithMods#31]
I upgraded my server from 0.10 directly to 0.11-hotfix1 and upon launching the server I got the notification that a few blocks were missing:

- betterwithmods:invisible_light
- betterwithmods:stoked_flame
- betterwithmods:kiln
- betterwithmods:waterwheel
- betterwithmods:invisible_light
- betterwithmods:windmill_block

I proceeded as normal but upon loading the world fully found out that level.dat was now donked up: http://pastebin.com/4vqBM4eY

I proceeded anyway and needless to say it is a good thing that Forge makes a world backup when missing blocks are detected and downgrades are a thing.

I'm not really sure if it helps but here's my fml-server-latest. I had to put it on my own hosting cause it's too big for pastebin; sorry if that limits the utility of it.
http://zantetsuken.com/minecraft/fml-server-latest.log

Missing blocks debacle starts at line 6351, then proceeds into the level.dat upset.

Let me know if I can provide any more info.

**Comment by primetoxinz** [comment]
in a recent refactor I removed the item form of those blocks which you should never get, losing these will harm nothing

**Comment by BeetoGuy** [comment]
Yeah, all that's gone is their registration in the item registry. Is there anything visibly different in the world?

**Comment by zcorcoran** [comment]
That's what I thought but then everything was shuffled up and my game crashed upon connecting, so I couldn't get a good look at the chaos. I think the problem is with the level.dat, but maybe that's just on my end, I'll mess around some more and see what's up. The server is a few forge versions behind so who knows.

**Comment by zcorcoran** [comment]
Ugh, sorry, I meant the SERVER crashed; ticking entity. I think because it forced usage of the backup level.dat and all the block IDs were wrong, so stuff was being cast to the wrong blocks. But again, the fact that it resorted to a backup level.dat could be a result of a problem on my end, which is what I'm gonna try to figure out.

**Comment by zcorcoran** [comment]
I don't know if it's relevant but I did notice this in line 30 of BlockInvisibleLight.java:

super(Material.AIR, "invisibleLight", null);

Everything else follows the pattern of super(Material.Stuff, "blockName"); without the "null", and the complaint Forge gives me about the level.dat is that betterwithmods:invisibleLight is registered as both Block{betterwithmods:invisibleLight} and Block{null}, so...

Sorry if I'm missing something obvious here. I can't find a lot of documentation offhand about how blocks are registered so I apologize if this is totally off-base.

**Comment by DaedalusGame** [comment]
that means BTWBlock will not register an item for it, as far as i can see

**Comment by zcorcoran** [comment]
I thought that might be the case, and it certainly does look like that.

I couldn't replicate this problem with the same modpack and a fresh world, so I'm forced to conclude it's this specific world's level.dat.

I downloaded the level.dat and removed the item forms of the aforementioned blocks manually with an NBT editor, because I'm a crazy person. I reuploaded the newly pruned level.dat and updated to 0.11 again, and there were no issues. Everything was in the right place, all blocks remained as they were, and Forge did not complain about level.dat.

Whatever caused the weird double registration appears to have been something else. Maybe the way Forge handles removing item IDs has some weird edge case that my particular level.dat happened to match.

Whatever it is, looks like it's not the mod. Thanks anyway and keep up the great work; BWM is the most fun I've had with modded MC in a long time.
